### Promise 学习小结——实例方法详解
***

#### 1、promise.then()
该方法返回一个promise。有两个参数，分别为promise在success和failure情况下的回调函数。

**语法**
```
p.then(onFulfilled, onRejected)
p.then((value) => {
    //resolved
}, (reason) => {
    //rejected
})
```
**参数**
> onFullfilled

一个function，当promise为fulfilled时调用，该函数有一个参数，为成功的返回值。

> onRejected

一个function，当promise为rejected时调用，该函数有一个参数，为失败的原因。

**示例**
```
const p = new Promise((resolve, reject) => {
    resolve('success');
    //or
    //reject('failed');
});
p.then(value => {
    console.log(value);//resolve
}, (reason) => {
    console.log(reason);//reject
});
```

#### 2、promise.catch()
该方法返回一个promise。catch方法只处理promise被拒绝的情况。

**语法**
```
p.catch(onRejected)
p.catch((reason) => {
    //rejected
})
```
**参数**
> onRejected

一个function，当promise为rejected时调用，该函数有一个参数，为失败的原因。

**描述**

catch方法主要用于处理promise组合

**示例**
```
const p = new Promise((resolve, reject) => {
    resolve('success');
    //or
    //reject('failed');
});
p.then(value => {
    console.log(value);//resolve
    throw 'oh no';
}).catch((reason) => {
    console.log(reason);//oh no
});
```
